feature(main): fix registry find manifest logic (#2314)
labring/sealos: docs/4.0/docs/examples/sealctl-registry.md

> # Using sealctl manager registry image

Reviewer: ```suggestion
# Using sealctl to manage registry images
```

> ## Image save to local registry dir

Reviewer: ```suggestion
## Save images to local registry directory
```

### default type

> default type can auto list images, using rootfs list images logic .ex charts,manifests,images dir.

Reviewer: Not sure how to change this right now. Any clarification?

```
registry images manager save to local dir by default type

Usage:
  sealctl registry save default [CONTEXT] [flags]

Flags:
      --arch string          pull images arch (default "arm64")
      --data-dir string      registry data dir path (default "/var/lib/registry")
  -h, --help                 help for default
      --max-pull-procs int   maximum number of goroutines for pulling (default 5)

Global Flags:
      --debug                   enable debug logger
      --root string             storage root dir (default "/var/lib/containers/storage")
      --runroot string          storage state dir (default "/run/containers/storage")
      --show-path               enable show code path
      --storage-driver string   storage-driver (default "overlay")
      --storage-opt strings     storage driver option
```

> #### How to using

Reviewer: Please replace `How to using` with `Usage` globally.

```suggestion
#### Usage
```

```shell
sealctl  registry  save default . 
```

### raw type

raw type using images as image list.

```
registry images manager save to local dir by raw type

Usage:
  sealctl registry save raw [flags]

Flags:
      --arch string          pull images arch (default "arm64")
      --data-dir string      registry data dir path (default "/var/lib/registry")
  -h, --help                 help for raw
      --images strings       images list
      --max-pull-procs int   maximum number of goroutines for pulling (default 5)

Global Flags:
      --debug                   enable debug logger
      --root string             storage root dir (default "/var/lib/containers/storage")
      --runroot string          storage state dir (default "/run/containers/storage")
      --show-path               enable show code path
      --storage-driver string   storage-driver (default "overlay")
      --storage-opt strings     storage driver option
```
#### How to using

```shell
sealctl  registry  save raw --images nginx
```

> ## Status registry

Reviewer: ```suggestion
## Check registry status
```

> need sealctl login or sealos login it.

Reviewer: ```suggestion
To use this command, you need to `sealctl login` or `sealos login` first.
```

```
registry status

Usage:
  sealctl registry status [flags]

Examples:
sealctl registry status

Flags:
  -h, --help   help for status
      --json   output in JSON format

Global Flags:
      --debug                   enable debug logger
      --root string             storage root dir (default "/var/lib/containers/storage")
      --runroot string          storage state dir (default "/run/containers/storage")
      --show-path               enable show code path
      --storage-driver string   storage-driver (default "overlay")
      --storage-opt strings     storage driver option
```

#### How to using

> ```

Reviewer: ```suggestion
```shell
```

> root@default-golang-0:~# sealctl  registry  status

Reviewer: I think things like `root@default-golang-0:~#` does not provide useful information and should be replaced with a cleaner `$` just to show that we're in the command line. We should not simply copy from our terminal and paste it into the docs without any modification.

```suggestion
$ sealctl registry status
```

+--------------------+-------------------------+---------+-------------------------------------------------------------------------+
| RegistryName       | ImageName               | Tag     | ImageID                                                                 |
+--------------------+-------------------------+---------+-------------------------------------------------------------------------+
| 192.168.64.63:5000 | kube-apiserver          | v1.19.0 | sha256:522d17d35a8994637d27d1232bebd35cfae8e3e21ab359431403f2b8023e332c |
| 192.168.64.63:5000 | kube-controller-manager | v1.19.0 | sha256:6c11a3d4d06385f7547a5ea0c3f0d5e7b12baa558111d01406ac1f778cb3f00b |
| 192.168.64.63:5000 | kube-proxy              | v1.19.0 | sha256:c752ecbd04bc4517168a19323bb60fb45324eee1e480b2b97d3fd6ea0a54f42d |
| 192.168.64.63:5000 | kube-scheduler          | v1.19.0 | sha256:529a1566960a5b3024f2c94128e1cbd882ca1804f222ec5de99b25567858ecb9 |
+--------------------+-------------------------+---------+-------------------------------------------------------------------------+
2023-01-01T12:31:49 info Image count 4
2023-01-01T12:31:49 info Images Version count 4
```

> filter logic:
> 
> filter key support: name, tag 
> 
> expressions:
> 
>   `*val`: suffix 
>   `val*`: prefix
>   `value`: equals
>   `<none>`: empty
>   `*val*`: contains

Reviewer: ```suggestion
### Filtering

Keys for `--filter`:
- `name`
- `tag`

Expressions:
- `*val`: suffix 
- `val*`: prefix
- `value`: equals
- `<none>`: empty
- `*val*`: contains
```

## Remove registry image 

```
registry rmi image

Usage:
  sealctl registry rmi [flags]

Examples:
sealctl registry  rmi labring/lvscare:v4.1.3

Flags:
  -h, --help          help for rmi
  -n, --name string   registry name (default "sealos.hub:5000")

Global Flags:
      --debug                   enable debug logger
      --root string             storage root dir (default "/var/lib/containers/storage")
      --runroot string          storage state dir (default "/run/containers/storage")
      --show-path               enable show code path
      --storage-driver string   storage-driver (default "overlay")
      --storage-opt strings     storage driver option
```

### How to using

```shell

> sealctl  registry  rmi 192.168.64.63:5000/kube-apiserver:v1.19.0 -n 192.168.64.63:5000

Reviewer: We should either use:

<pre>
```shell
$ sealctl <commands> ...
```
</pre>

Or:

<pre>
```shell
sealctl <commands> ...
```
</pre>

And should not use them both.

```

## Prune registry

> Must be using in registry node, using binary deploy mode.Not support container deploy.

Reviewer: ```suggestion
This command can only be run on a registry node, which used the binary deploy mode. Does not support nodes using container deploy mode.
```
